### ISSUE TITLE
Output mesh seems wrong

### ISSUE THREAD

**Issue by lihongyi1234** [sdmiller/cpu_tsdf#57]
![image](https://github.com/sdmiller/cpu_tsdf/assets/36233282/f0ab370c-3775-40f4-b748-3e53ea9f7147)
I run in Window platform, but output mesh seems wrong, vertex always (-6.0029 -6.0029 -6.0029).

**Comment by lihongyi1234** [comment]
@sdmiller have you executable  ./integrate with sample data, I tested it using default option, result is nothing

**Comment by sdmiller** [comment]
Just an update: @lihongyi1234 and I have been speaking via email, and this bug was found under the following conditions:
* Windows 10
* VS2019
* PCL 1.12

I've just tried to reproduce the same behavior on my system (MacOS 12.3) with the same PCL version (1.12), but the output works as expected. I'm afraid this is very likely something specific to Windows; perhaps PCL's PLY writer and/or Marching Cubes implementation in Windows.

I'll try to get a Windows VM set up at some point. In the meantime, the easiest way to debug this will be if you save the full TSDF output from your `integrate` executable via the `--save-tsdf` options I just added [in this pull request](https://github.com/sdmiller/cpu_tsdf/pull/58). That will let me try to do the meshing on my side, to see if this is an issue with my TSDF logic, or with something on the Marching Cubes and/or IO side.
